<launch>
  <!-- These are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="with_gripper" default="true"/>
  <arg name="use_platform" default="false"/>
  <arg name="use_sim" default="true"/>
  <arg name="use_rviz" default="false"/>
  <arg name="usb_port" default="/dev/ttyUSB0"/>
  <arg name="arm_id" default="1"/>
  <arg name="cmd_op" default="false"/>

  <arg name="gui" default="true"/>
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="false"/>

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <group if="$(arg use_sim)">
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="gui" value="$(arg gui)"/>
      <arg name="paused" value="$(arg paused)"/>
      <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    </include>
  </group>

  <!-- Launch Rviz -->
  <group if="$(arg use_rviz)">
    <include file="$(find open_manipulator_p_description)/launch/open_manipulator_p_rviz.launch">
        <arg name="with_gripper" value="$(arg with_gripper)"/>
    </include>
  </group>

  <include file="$(find papras)/launch/one_arm.launch">
    <arg name="use_platform" value="$(arg use_platform)"/>
    <arg name="use_sim" value="$(arg use_sim)"/>
    <arg name="with_gripper" value="$(arg with_gripper)"/>
    <arg name="usb_port" value="$(arg usb_port)"/>
    <arg name="arm_id" value="$(arg arm_id)"/>
    <arg name="synced_gui" value="$(arg cmd_op)"/>
  </include>
</launch>
